rpcclient/wallet.go

```diff
@@ -798,9 +798,14 @@ func (c *Client) PurchaseTicketAsync(fromAccount string,
 		expiryVal = *expiry
 	}
 
+	noSplitTx := true
+	if noSplitTransaction != nil {
+		noSplitTx = *noSplitTransaction
+	}
+
 	cmd := dcrjson.NewPurchaseTicketCmd(fromAccount, spendLimit.ToCoin(),
 		&minConfVal, &ticketAddrStr, &numTicketsVal, &poolAddrStr,
-		&poolFeesFloat, &expiryVal, nil, noSplitTransaction)
+		&poolFeesFloat, &expiryVal, nil, &noSplitTx)
```

Comment:
Could you avoid the marshaling issue by using `dcrjson.String("")` instead of `nil`?

Reply:
I hit a similar bug recently and the cleanest solution was to not pass `nil` pointers.  https://github.com/decred/dcrrpcclient/pull/89/files

Reply:
The nil passed in that code is there by default. Area of concentration is the no split  variable which can never be nil. Its fine now

Reply:
I understand what you are trying todo, also that nil is there already, but the problem would be solved if we quit passing `nil` pointers because it confuses the code downstream.

Reordering the `PurchaseTicketCmd` fields is just working around the bug.  Give json.Marshal something to dereference instead of a nil pointer and the code isn't broken.

Say there were other fields where the passed values were nil, would we expect to reorder the struct fields to make it work?  No, we would not pass a nil pointer, and instead pass a valid pointer to a zero value.

Reply:
you're absolutely right. Ran some tests. Turns out drcjson marshaling halts on encountering a nil pointer